### ISSUE TITLE
Wiki Request - New Articles on Java are required

### ISSUE THREAD

**Issue by koustuvsinha** [FreeCodeCamp/wiki#534]
This is an epic issue to track the development of Java Wiki articles. The target is to have smaller articles on individual topics so that they can be easily referenced by @camperbot . 

These are our requirements. Feel free to add comment on articles which are needed and I will update it here. If you want to work on a specific article, comment here and let others know. Also, refer this issue in subsequent PR's for easy tracking.

- [x] [Home Page](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Java)
- [x] [Introduction](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Java-Introduction)
- [ ] Basics - In progress #526 
- [ ] Data Types
- [ ] Control Statements
- [ ] Loops
- [ ] Constructors
- [ ] Inheritance
- [ ] Interfaces
- [ ] Abstract Classes
- [ ] Packages
- [ ] Lambda Expressions
- [ ] Project Structure
- [ ] Nested Inner Anonymous Class
- [ ] Unit Testing in Java
- [ ] Web Development with Java
- [ ] Getting started with Spring
- [ ] Spring Installation
- [ ] Planning a Spring Web App
- [ ] Spring Model Layer
- [ ] Spring Routes
- [ ] Spring Views
- [ ] Spring Templates
- [ ] Testing in Spring
- [ ] Spring Refactoring & Improvements
- [ ] Java More Resources

@Rafase282 @raisedadead please add relevant labels as per other issues. @alayek have a look.

**Comment by alayek** [comment]
Great job @koustuvsinha !

**Comment by ScrumUser1** [comment]
I would like to contribute to the loop article for java. Can we talk more about this so i know how you would want it to be?

**Comment by alayek** [comment]
@ScrumUser1 yes! We communicate via gitter. Kindly contact me or @koustuvsinha via hitter, and one of us would discuss with you how the article would be.

**Comment by koustuvsinha** [comment]
@ScrumUser1 can you work on a different article as another user has already submitted a PR on loops in #560 ? If you need any help please feel free to contact me in [Gitter](https://gitter.im/koustuvsinha).

**Comment by trishulpani** [comment]
I'd like to take up Constructors. Let me know if anyone else is looking into it.

**Comment by Shaif00** [comment]
Hii, I'm writing data types and control Statements, will create a pull req soon.

**Comment by alayek** [comment]
@koustuvsinha FYI. Let me know if I need to assign something.

**Comment by ilovejava1988** [comment]
I would like to take up abstract classes.

**Comment by AakanxuShah** [comment]
@trishulpani  : If you are not doing Constructors , can I write an article about it ?

**Comment by trishulpani** [comment]
@AakanxuShah - Apologies for the late reply. I have updated the article for Java-Constructors. Please take a look and feel free to add to it. Thanks!

**Comment by koustuvsinha** [comment]
hey @ilovejava1988 ! are you still up for taking abstract classes? :) Please send us a PR. If you have any doubts, talk with us in our [Gitter channel](https://gitter.im/FreeCodeCamp/Wiki)

**Comment by ilovejava1988** [comment]
Hi @koustuvsinha Yup i am working on it.i will be sending the PR in a day.

**Comment by trishulpani** [comment]
Hi @ilovejava1988 / @koustuvsinha  , I had started work on Abstract classes. If you are not already done with the article, maybe you can take a look at [Java Abstract Classes](https://github.com/trishulpani/wiki/blob/master/Java-Abstract-Classes.md). Let me know if you want me to leave it as it is or create a PR.

**Comment by shailu0827** [comment]
I can write up articles for abstract classes and collections topics, please let me know the guidelines or provide a link where the guidelines are?
